packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts

```diff
@@ -0,0 +1,185 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {
+    Detector,
+    Resource,
+    CONTAINER_RESOURCE,
+    K8S_RESOURCE,
+    ResourceDetectionConfigWithLogger,
+  } from '@opentelemetry/resources';
+  import * as https from 'https';
+  import * as fs from 'fs';
+  import * as util from 'util';
+
+  /**
+ * The AwsEksDetector can be used to detect if a process is running in AWS Elastic
+ * Kubernetes and return a {@link Resource} populated with data about the Kubernetes
+ * plugins of AWS X-Ray. Returns an empty Resource if detection fails.
+ *
+ * See https://docs.amazonaws.cn/en_us/xray/latest/devguide/xray-guide.pdf
+ * for more details about detecting information for Elastic Kubernetes plugins
+ */
+  
+export class AwsEksDetector implements Detector {
+    readonly K8S_SVC_URL = "https://kubernetes.default.svc";
+    readonly K8S_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+    readonly K8S_CERT_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+    readonly AUTH_CONFIGMAP_PATH ="/api/v1/namespaces/kube-system/configmaps/aws-auth";
+    readonly CW_CONFIGMAP_PATH = "/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info";
+    readonly CONTAINER_ID_LENGTH = 64;
+    readonly DEFAULT_CGROUP_PATH = "/proc/self/cgroup";
+    readonly MILLISECOND_TIME_OUT = 2000;
+
+    private static readFileAsync = util.promisify(fs.readFile);
+    private static fileAccessAsync = util.promisify(fs.access);
+  
+    async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
+      try {
+        await AwsEksDetector.fileAccessAsync(this.K8S_TOKEN_PATH);
+        await AwsEksDetector.fileAccessAsync(this.K8S_CERT_PATH);
+        
+        if (!this._isEks(config)) {
+          config.logger.debug('AwsEcsDetector failed: Process is not running on Eks');
+          return Resource.empty();
+        }
+      
+        const containerId = await this._getContainerId(config);
+        const clusterName = await this._getClusterName(config);
+
+        return !containerId && !clusterName
+        ? Resource.empty()
+        : new Resource({
+          [K8S_RESOURCE.CLUSTER_NAME]: clusterName || '',
+          [CONTAINER_RESOURCE.ID]: containerId || '',
+        });
+      } catch (e) {
+        config.logger.warn('Not running on K8S');
+        return Resource.empty();
+      }
+    }
+
+    private async _isEks(config: ResourceDetectionConfigWithLogger): Promise<boolean> {
+      const options = {
+        hostname: this.K8S_SVC_URL,
+        path: this.AUTH_CONFIGMAP_PATH,
+        method: 'GET',
+        timeout: this.MILLISECOND_TIME_OUT,
+        HEADERS: {
+          "Authorization" : this._getK8sCredHeader(config),
+        },
+        ca: JSON.stringify(AwsEksDetector.readFileAsync(this.K8S_CERT_PATH)),
```

Comment:
Let's only read the CA file once

Reply:
Also I think this is stringifying a promise, the unit test might be missing a sanity check of the content of this option

Reply:
I suppose there shouldn't be a need to call JSON.stringify for the cert anyways

Reply:
After looking into nock documentation, there doesn't seem to be a plausible way nock for a sanity check for the content of this option. Are there any alternatives?